services/QuillLMS/engines/evidence/app/models/evidence/research/gen_ai/trial.rb

```diff
@@ -76,22 +78,25 @@ def average_g_eval_score
         def scores = g_evals&.values&.flatten&.compact&.map(&:to_f)
 
         def run
```

Comment:
You might consider extracting this `run` method and its support methods out of the model and into a service or worker.